spec/api-web-contents-spec.js

```diff
@@ -98,6 +98,114 @@ describe('webContents module', function () {
     })
   })
 
+  describe('before-input-event event', () => {
+    it('can prevent document keyboard events', (done) => {
+      w.loadURL('file://' + path.join(__dirname, 'fixtures', 'pages', 'key-events.html'))
+      w.webContents.once('did-finish-load', () => {
+        let error = null
+        const onEvent = (event) => {
+          error = new Error('Unexpected keydown message')
+        }
+        ipcMain.on('keydown', onEvent)
+
+        let beforeInputEventReceived = false
+        w.webContents.on('before-input-event', (event, input) => {
+          assert.equal(input.type, 'keyDown')
+          assert.equal(input.key, 'a')
+          beforeInputEventReceived = true
+          event.preventDefault()
+        })
+
+        w.webContents.sendInputEvent({type: 'keyDown', keyCode: 'a'})
+
+        setTimeout(() => {
+          assert.ok(beforeInputEventReceived)
+          ipcMain.removeListener('keydown', onEvent)
+          done(error)
+        }, 500)
```

Comment:
Timeouts can be pretty flaky on CI, maybe it could just send two input events instead, prevent the first and then call `done()` when the second is delivered in the page?